# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [4]:
# Your code here
import pandas as pd

df = pd.read_csv('../data/car-sales.csv')

print("Number of lines and rows: ", df.shape)

Number of lines and rows:  (10, 5)


### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [26]:
# Your code here
print(df)

for i in range(0, df.dtypes.size):
    if df.dtypes[i] == 'object':
        print(df.columns[i] + ": " + "dinh tinh")
    else:
        print(df.columns[i] + ": " + "dinh luong")

  Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0
Manufacturer: dinh tinh
Colour: dinh tinh
Odometer: dinh luong
Doors: dinh luong
Price: dinh luong


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [7]:
# Your code here
print("Number of cars: ", df.shape[0])

Number of cars:  10


### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [12]:
# Your code here
df['Manufacturer'].value_counts()

Toyota    4
Honda     3
Nissan    2
BMW       1
Name: Manufacturer, dtype: int64

### 5. Đếm số xe theo màu


In [14]:
# Your code here
df['Colour'].value_counts()

White    4
Blue     3
Red      1
Black    1
Green    1
Name: Colour, dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [16]:
# Your code here
df['Price'].sum()

76450.0

### 7. Top 3 xe có giá cao nhất


In [17]:
# Your code here
df.nlargest(3, 'Price')

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [18]:
# Your code here
df.nlargest(3, 'Odometer')

Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [115]:
# Your code here
print("Ti le xe Honda:", df[df['Manufacturer'] == 'Honda'].shape[0] / df.shape[0])
print("Gia trung binh xe honda", df[df['Manufacturer'] == 'Honda']['Price'].mean())
print(df[df['Manufacturer'] == 'Honda']['Colour'].unique())

Ti le xe Honda: 0.3
Gia trung binh xe honda 6500.0
['Red' 'Blue']


# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [57]:
# Your code here
print("Gia trung binh cua cac xe: ", df['Price'].mean())
print("Gia tri trung vi cac xe: ", df['Price'].median())
print("Mien gia tri cua xe: ", df['Price'].max() - df['Price'].min())
print("Mien phan vi cua xe: ", df['Price'].quantile(0.75) - df['Price'].quantile(0.25))
print("Diem phan vi 0.4 va 0.6", [df['Price'].quantile(0.4), df['Price'].quantile(0.6)])

Gia trung binh cua cac xe:  7645.0
Gia tri trung vi cac xe:  6625.0
Mien gia tri cua xe:  18500.0
Mien phan vi cua xe:  2750.0
Diem phan vi 0.4 va 0.6 [5750.0, 7000.0]


---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [5]:
# Your code here
tips_df = pd.read_csv('../data/tips.csv')

2. Xoá cột `Unnamed: 0`


In [6]:
# Your code here
tips_df.drop(['Unnamed: 0'], axis=1, inplace=True)

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [71]:
# Your code here
print("So dong:", tips_df.shape[0])
print("So thuoc tinh:", tips_df.shape[1])

So dong: 244
So thuoc tinh: 7


4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [74]:
# Your code here
print(tips_df.info())

for i in range(0, tips_df.dtypes.size):
    if tips_df.dtypes[i] == 'object':
        print(tips_df.columns[i] + ": " + "dinh tinh")
    else:
        print(tips_df.columns[i] + ": " + "dinh luong")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB
None
total_bill: dinh luong
tip: dinh luong
sex: dinh tinh
smoker: dinh tinh
day: dinh tinh
time: dinh tinh
size: dinh luong


5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [9]:
# Your code here
tips_df['sex'].value_counts() / tips_df.shape[0]

Male      0.002637
Female    0.001461
Name: sex, dtype: float64

6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [97]:
# Your code here
num_of_smokers = tips_df[tips_df['smoker'] == 'Yes'].shape[0]

print("Co " + str(num_of_smokers) + " nguoi hut thuoc, chiem ti le " + str(round(num_of_smokers / tips_df.shape[0], 2)))

Co 93 nguoi hut thuoc, chiem ti le 0.38


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [114]:
# Your code here
best_day = tips_df['day'].value_counts().nlargest(1)

print("Ngay khach hang den nhieu nhat la " + str(best_day.index[0]) + ", chiem ti le " + str(round(best_day[0] / tips_df.shape[0], 2)))

Ngay khach hang den nhieu nhat la Sat, chiem ti le 0.36


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [16]:
# Your code here
best_time = tips_df['time'].value_counts().idxmax()

print("Nha hang phuc vu bua an " + best_time + " nhieu nhat")

Nha hang phuc vu bua an Dinner nhieu nhat


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [121]:
# Your code here
largest_size = tips_df['size'].max()

print("Nhom khach hang dung bua tai nha hang nhieu nhat la " + str(largest_size) + " nguoi")

Nhom khach hang dung bua tai nha hang nhieu nhat la 6 nguoi


10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [143]:
# Your code here
def round_numbers(x):
    return round(x)

average_size_by_time = tips_df.groupby('time')['size'].mean().apply(round_numbers)

average_size_by_time

time
Dinner    3
Lunch     2
Name: size, dtype: int64

---